frontend/src/components/HeaderNav.tsx

```diff
@@ -1,19 +1,70 @@
-import {AppBar, Avatar, Box, Container, IconButton, Toolbar, Typography} from "@mui/material";
+import {
+    AppBar,
+    Avatar,
+    Box,
+    Container,
+    IconButton,
+    ListItemIcon,
+    Menu,
+    MenuItem,
+    Toolbar,
+} from "@mui/material";
 import React from "react";
+import {useUserContext} from "../context/userContext";
+import {Logout as LogoutIcon} from "@mui/icons-material";
+import {UserApi} from "../api/mixify-api";
+import {toast} from "react-toastify";
+import {useNavigate} from "react-router-dom";
+import logo from "../logo.png";
 
 export default function HeaderNav() {
+    const {user} = useUserContext();
+    const navigate = useNavigate();
+
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+    const open = Boolean(anchorEl);
+
+    const onLogout = async () => {
+        await UserApi.logout();
+
+        toast.success("Successfully logged out.");
+
```

Comment:
If I understand this correctly, might want to do a setUser(null), where setUser comes from the useUserContext()

Reply:
Good point 👍  I'll change that

Reply:
Done